[MIG] stock_manual_transfer: Migration to 17.0 T#79454

xmglord opened the pull request.

Commit:
[ADD] stock_manual_transfer: New module added

- New module added in order to allow the user to create stock transfers
  manually.

carralc committed:
[FIX] stock_manual_transfer: remove .unlink() method override

Due to upstream updates adding support for @api.ondelete and
pre-commit-vauxoo detecting the override of .unlink() method as
a code smell, this transforms the old logic into more
idiomatic odoo code.

carralc committed:
[FIX] stock_manual_transfer: super().unlink() was not removed in original commit

While the original fix (https://github.com/Vauxoo/addons-vauxoo/commit/cbd70ffe5b6693e9ecd299e3c935a3dcfa4a0131)
did fix the immediate lint messages, we didn't actually stop it from
calling super().unlink(), thus it didn't actually fix what it intended
to fix. Here we remove that call to super().unlink() so the issue is
properly fixed.

xmglord commented on stock_manual_transfer/views/stock_manual_transfer_views.xml:
```diff
@@ -33,26 +33,26 @@
                                     <field name="sequence" widget="handle" />
                                     <field name="product_id" />
                                     <field name="product_uom_qty" />
-                                    <field name="product_uom_id" groups="uom.group_uom" optional="show" />
```
This change and the one below are made because if the user don't belong to that group the compute of that field is not triggered, do you have any context why this happens in v17? @luisg123v

xmglord commented:
@luisg123v please review.

luisg123v commented on stock_manual_transfer/i18n/es.po:
```diff
@@ -0,0 +1,420 @@
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Odoo Server 15.0+e\n"
```
Please regenerate this file.

xmglord replied:
done

luisg123v commented on stock_manual_transfer/models/stock_route.py:
```diff
@@ -0,0 +1,7 @@
+from odoo import fields, models
+
+
+class StockLocationRoute(models.Model):
```
Rename also class

xmglord replied:
done

luisg123v commented on stock_manual_transfer/views/stock_manual_transfer_views.xml:
```diff
@@ -0,0 +1,105 @@
+                                    <field name="sequence" widget="handle" />
+                                    <field name="product_id" />
+                                    <field name="product_uom_qty" />
+                                    <field name="product_uom_id" optional="show" />
```
Why?

xmglord replied:
Because if the user don't belong to that group the compute of that field was not triggered.

xmglord replied:
Because the tests fail but know I think that is a problem with the tests because functionally it works. Why you think the test may be failing?

luisg123v replied:
What is the error message?

You should probably add this same file without group but invisible

luisg123v commented on stock_manual_transfer/views/stock_manual_transfer_views.xml:
```diff
@@ -0,0 +1,105 @@
+                        </h1>
+                    </group>
+                    <group col="4">
+                        <field name="warehouse_id" options="{'no_create': True}" readonly="state != 'draft'" />
```
Why this?

The state attribute was not here.

If it's due to the Python attribute not being considered anymore, then you should remove that attribute from the field declaration.

xmglord replied:
Done, thanks for the info.

luisg123v reviewed (changes requested):
Commit description says:

> Update fields names and models names with the respective file name.

- That is not clear enough IMHO
- Are any field actually being updated?

allanpa88 committed:
[MIG] stock_manual_transfer: Migration to 16.0 #81066

- Changed name for deprecated models like stock.location.route to
  stock.route and stock move to Operations(stock.move.line) to Stock
  Moves.
- Added user to group uom to fix unit tests and avoid error for not
  null violation in product uom.

luisg123v commented on stock_manual_transfer/models/__init__.py:
```diff
@@ -1,4 +1,4 @@
 from . import procurement_group
-from . import stock_location_route
+from . import stock_route
```
These imports are sorted alphabetically, let's keep that order.

allanpa88 committed:
[IMP] stock_manual_transfer:  Add domain to route_id field

Add domain to route_id field to show the only the routes that are
supplied_wh_id = warehouse_id. This is to avoid
inconsistencies between the rule and the warehouse.

allanpa88 committed:
[FIX] stock_manual_transfer: Prevent creation of duplicate transfers

To prevent any potential duplication, this fix introduces a
return-early pattern. Now, transfers are not created if the manual
transfer record has already been published.

Issue reported on helpdesk ticket #25879

allanpa88 committed:
[FIX] stock_manual_transfer: Avoid Unit of Measure validation error without group

A validation error is raised when a user without the "Manage Multiple
Units of Measure" access tries to save or validate a manual transfer.

This fix allows any user to create manual transfers, but only users with
"Units of Measure" access can modify this attribute on the lines.

allanpa88 committed:
[IMP] stock_manual_transfer: Rename model Manual Transfer Line

Renamed the model 'stock.manual_transfer_line' to
'stock.manual_transfer.line' for consistency with previous
versions of this model.

Commit:
[MIG] stock_manual_transfer: Migration to 17.0

- Update use of quantity as part of [1].
- Simplify the way to define modifiers as states, required, readonly,
  invisible and column_invisible as part of [2].
- Fix tests.

[1] odoo/odoo#137864
[2] odoo/odoo#104741

xmglord commented:
@luisg123v already rebased and fixed.

luisg123v commented on stock_manual_transfer/__manifest__.py:
```diff
@@ -0,0 +1,28 @@
+    "website": "https://www.vauxoo.com",
+    "license": "LGPL-3",
+    "category": "Inventory/Inventory",
+    "version": "17.0.1.0.1",
```
```suggestion
    "version": "17.0.1.0.0",
```

xmglord replied:
done

luisg123v reviewed (approved):
LGTM :+1:
